src/searchbar.cpp

```diff
@@ -75,13 +77,37 @@ SearchBar::SearchBar(QWidget *parent) :
     QString style(byteContent);
     m_completer.popup()->setStyleSheet(style);
 
-    connect(this, &QLineEdit::textEdited, this, &SearchBar::updateCompletion);
+    SuggestionListWorker *worker = new SuggestionListWorker;
+    worker->moveToThread(&m_suggestionThread);
+    connect(&m_suggestionThread, &QThread::finished, worker, &QObject::deleteLater);
+    connect(this, &SearchBar::searchSuggestion, worker, &SuggestionListWorker::searchSuggestions);
+    connect(worker, &SuggestionListWorker::suggestionTitle, this, [=](const QString& title) {
+        m_suggestionsList.append(title);
+    });
+    connect(worker, &SuggestionListWorker::suggestionUrl, this, [=](const QUrl& url) {
+        m_urlList.append(url);
+    });
```

Comment:
It is better to have only one signal here to ensure that both `m_suggestionList` and `m_urlList` are sync.
There is no reason that title and url are "sent" separatly.